### PR TITLE
JavaScript Tips & Tricks / Singleton Classes

### PR DESCRIPTION
**Pull request by gomorizsolt**
Issue #39.

I guess that's what we ended up with.

### CODE REVIEW

File: js/tips-tricks.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# JavaScript / Tips & Tricks
+
+#### Singleton classes
+
+```js
+class Foo {
+    // ...
+}
+
+class Singleton {
```
Comment (thisismydesign) [inline js/tips-tricks.md:10]:
```suggestion
export default class Singleton {
```

File: js/tips-tricks.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# JavaScript / Tips & Tricks
+
+#### Singleton classes
+
+```js
+class Foo {
```
Comment (thisismydesign) [inline js/tips-tricks.md:6]: Let's use a real example we have elsewhere:
```suggestion
import ElectronStore from "electron-store";
```

File: js/tips-tricks.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      }
+
+      return this.instance;
+   }
+}
+
+const s1 = Singleton.getInstance();
+const s2 = Singleton.getInstance();
+
+console.log(s1 === s2); // => true
```
Comment (thisismydesign) [inline js/tips-tricks.md:25]: IMO this and the instantiations are not needed, should be obvious.

Comment (gomorizsolt) [reply js/tips-tricks.md:25]: Makes sense, thanks! After all that's the gist of singleton classes.

Comment (gomorizsolt) [reply js/tips-tricks.md]: https://github.com/c-hive/guides/pull/43/commits/8fed5a0eedb9b72041c7049afcd2709a5f4354a6

File: js/tips-tricks.md (hunk 2 above)
Comment (gomorizsolt) [reply js/tips-tricks.md]: https://github.com/c-hive/guides/pull/43/commits/8fed5a0eedb9b72041c7049afcd2709a5f4354a6

File: js/tips-tricks.md (hunk 1 above)
Comment (gomorizsolt) [reply js/tips-tricks.md]: https://github.com/c-hive/guides/pull/43/commits/8fed5a0eedb9b72041c7049afcd2709a5f4354a6

**Pull request merged**